File: Basic/tutorial-applestamp/contracts/src/test/java/com/tutorial/contracts/ContractTests.java
package com.tutorial.contracts;

import com.tutorial.states.AppleStamp;
import com.tutorial.states.TemplateState;
import net.corda.core.contracts.UniqueIdentifier;
import net.corda.core.identity.CordaX500Name;
import net.corda.testing.core.TestIdentity;
import net.corda.testing.node.MockServices;
import org.junit.Test;

import java.util.Arrays;

import static net.corda.testing.node.NodeTestUtils.ledger;


public class ContractTests {
    private final MockServices ledgerServices = new MockServices(Arrays.asList("com.tutorial.contracts"));
    TestIdentity alice = new TestIdentity(new CordaX500Name("Alice",  "TestLand",  "US"));
    TestIdentity bob = new TestIdentity(new CordaX500Name("Alice",  "TestLand",  "US"));

    //Template Tester
    @Test
    public void issuerAndRecipientCannotHaveSameEmail() {
        TemplateState state = new TemplateState("Hello-World",alice.getParty(),bob.getParty());
        ledger(ledgerServices, l -> {
            l.transaction(tx -> {
                tx.input(TemplateContract.ID, state);
                tx.output(TemplateContract.ID, state);
                tx.command(alice.getPublicKey(), new TemplateContract.Commands.Send());
                return tx.fails(); //fails because of having inputs
            });
            l.transaction(tx -> {
                tx.output(TemplateContract.ID, state);
                tx.command(alice.getPublicKey(), new TemplateContract.Commands.Send());
                return tx.verifies();
            });
            return null;
        });
    }

    //Basket of Apple cordapp testers
    @Test
    public void StampIssuanceCanOnlyHaveOneOutput(){
        AppleStamp stamp = new AppleStamp("FUji4072", alice.getParty(),bob.getParty(),new UniqueIdentifier());
        AppleStamp stamp2 = new AppleStamp("HoneyCrispy7864", alice.getParty(),bob.getParty(),new UniqueIdentifier());

        ledger(ledgerServices, l -> {
            l.transaction(tx -> {
                tx.output(AppleStampContract.ID, stamp);
                tx.output(AppleStampContract.ID, stamp2);
                tx.command(alice.getPublicKey(), new AppleStampContract.Commands.Issue());
                return tx.fails(); //fails because of having inputs
            });
            l.transaction(tx -> {
                tx.output(AppleStampContract.ID, stamp);
                tx.command(alice.getPublicKey(), new AppleStampContract.Commands.Issue());
                return tx.verifies();
            });
            return null;
        });
    }

    @Test
    public void StampMustHaveDescription(){
        AppleStamp stamp = new AppleStamp("", alice.getParty(),bob.getParty(),new UniqueIdentifier());
        AppleStamp stamp2 = new AppleStamp("FUji4072", alice.getParty(),bob.getParty(),new UniqueIdentifier());

        ledger(ledgerServices, l -> {
            l.transaction(tx -> {
                tx.output(AppleStampContract.ID, stamp);
                tx.command(Arrays.asList(alice.getPublicKey(),bob.getPublicKey()), new AppleStampContract.Commands.Issue());
                return tx.fails(); //fails because of having inputs
            });
            l.transaction(tx -> {
                tx.output(AppleStampContract.ID, stamp2);
                tx.command(Arrays.asList(alice.getPublicKey(),bob.getPublicKey()), new AppleStampContract.Commands.Issue());
                return tx.verifies();
            });
            return null;
        });
    }


}

File: java-samples/Basic/tutorial-jarsigning/contracts/src/test/kotlin/com/tutorial/contracts/ContractTests.kt
package com.tutorial.contracts

import com.tutorial.states.AppleStamp
import net.corda.core.identity.CordaX500Name
import net.corda.testing.core.TestIdentity
import net.corda.testing.node.MockServices
import net.corda.testing.node.ledger
import org.junit.Test
import com.tutorial.states.TemplateState
import net.corda.core.contracts.UniqueIdentifier
import java.security.PublicKey
import java.util.*

class ContractTests {
    private val ledgerServices: MockServices = MockServices(listOf("com.tutorial"))
    var alice = TestIdentity(CordaX500Name("Alice", "TestLand", "US"))
    var bob = TestIdentity(CordaX500Name("Alice", "TestLand", "US"))

    @Test
    fun issuerAndRecipientCannotHaveSameEmail() {
        val state = TemplateState("Hello-World", alice.party, bob.party)
        ledgerServices.ledger {
            // Should fail bid price is equal to previous highest bid
            transaction {
                //failing transaction
                input(TemplateContract.ID, state)
                output(TemplateContract.ID, state)
                command(alice.publicKey, TemplateContract.Commands.Create())
                fails()
            }
            //pass
            transaction {
                //passing transaction
                output(TemplateContract.ID, state)
                command(alice.publicKey, TemplateContract.Commands.Create())
                verifies()
            }
        }
    }

    @Test
    fun stampIssuanceCanOnlyHaveOneOutput(){
        val stamp = AppleStamp("FUji4072", alice.party, bob.party, UniqueIdentifier())
        val stamp2 = AppleStamp("HoneyCrispy7864", alice.party, bob.party, UniqueIdentifier())

        ledgerServices.ledger {
            // Should fail bid price is equal to previous highest bid
            transaction {
                //failing transaction
                output(AppleStampContract.ID, stamp)
                output(AppleStampContract.ID, stamp2)
                command(alice.publicKey, AppleStampContract.Commands.Issue())
                fails()
            }
            //pass
            transaction {
                //passing transaction
                output(AppleStampContract.ID, stamp)
                command(alice.publicKey, AppleStampContract.Commands.Issue())
                verifies()
            }
        }
    }

    @Test
    fun StampMustHaveDescription(){
        val stamp = AppleStamp("", alice.party, bob.party, UniqueIdentifier())
        val stamp2 = AppleStamp("FUji4072", alice.party, bob.party, UniqueIdentifier())

        ledgerServices.ledger {
            // Should fail bid price is equal to previous highest bid
            transaction {
                //failing transaction
                output(AppleStampContract.ID, stamp)
                command(Arrays.asList(alice.publicKey, bob.publicKey), AppleStampContract.Commands.Issue())
                fails()
            }
            //pass
            transaction {
                //passing transaction
                output(AppleStampContract.ID, stamp2)
                command(Arrays.asList(alice.publicKey, bob.publicKey), AppleStampContract.Commands.Issue())
                verifies()
            }
        }
    }





}